prospector/commit_processor/feature_extractor.py

```diff
@@ -74,7 +74,7 @@ def extract_changed_relevant_paths(
     relevant_paths = []
     for advisory_path in advisory_record.paths:
         relevant_paths += filter(
-            lambda path: advisory_path in path, commit.changed_files
```

Comment:
You replaced the filtering logic with a more strict version. Hence the expected values will be faulty, i.e. water-bending will not match. I used the `in` instead of `endswith` because I could imagine the following scenarios.

1. In a comment someone will only mention the name of the file without the extension, in which case the `endswith` won't find it. 
2. Someone may mention a class or module in a comment, which has the same name as the containing *.py file but it will not have the .py extension. In this case, `endswith` could not match either.